src/distilabel/steps/tasks/typing.py

```diff
@@ -15,18 +15,57 @@
 from typing import Any, Dict, List, Literal, Optional, Tuple, Type, Union
 
 from pydantic import BaseModel
-from typing_extensions import TypedDict
+from typing_extensions import Required, TypedDict
 
 
 class ChatItem(TypedDict):
-    role: str
+    role: Literal["system", "user", "assistant"]
     content: str
```

Comment:
Let's update the type of `content` to something like `Union[str, TextContent, ImageContent]` instead of creating more types like `ConversationType`